internal/edge/stack.go

```diff
@@ -278,7 +292,9 @@ func (manager *StackManager) deleteStack(ctx context.Context, stack *edgeStack,
 		return
 	}
 
+	manager.mu.Lock()
```

Comment:
why here we don't put Lock at the beginning of the method with defer Unlock, like in `deployStack` ?

Reply:
Because locking is not free :-)

Reply:
wdym by locking is not free? Shouldn't the unlock be called anyway if it is deferred ?

Reply:
I mean that there is a cost to pay if we lock for longer than necessary (higher contention), in addition to the deferred call. This case is simple enough to reason about and there are no earlier returns to make it trickier.